### DISCUSSION TITLE
How can I make the form display horizontally?

### DISCUSSION THREAD

**Discussion by junmaqiang** [fyne-io/fyne#4258] [Q&A] [answered]
# I want to  this  Effect

![20210326134506755](https://github.com/fyne-io/fyne/assets/1819457/7aa5b57a-e847-4bcf-bbab-54b9696c12ba)

**Comment by Jacalz** [comment, marked as answer]
The form widget does not support that layout option at the moment

**Reply by Jacalz** [reply]
I suppose you can hack around it by putting two form widgets with a single item next to each other

**Reply by junmaqiang** [reply]
It would be perfect if you could provide parameters to select horizontal or vertical orientation.

**Comment by andydotxyz** [comment]
If you want to request this feature please use a feature request :)

**Comment by junmaqiang** [comment]
@Jacalz  
Can you give me a code reference? Thank you very much.

```
	me := widget.NewEntry()
	me.SetPlaceHolder("KwdMAjGmerYanjeui5")
	meCon := container.NewHBox(me)

	to := widget.NewEntry()
	to.SetPlaceHolder("100")
	toCon := container.NewHBox(to)

	formMe := &widget.Form{
		Items: []*widget.FormItem{
			{Text: "from", Widget: meCon, HintText: ""},
			{Text: "to", Widget: toCon, HintText: ""},
		},
		OnCancel: func() {
			fmt.Println("cong")
		},
		OnSubmit: func() {
			fmt.Println("cong Form submitted")
		},
	}

```

**Reply by Jacalz** [reply]
Just but two form widgets (with one form item each) next to each other in a grid for example. Like I said, that is the only workaround for now.

I don't know why thou add HBox components in the code above. It will just make the entry small.

**Reply by junmaqiang** [reply]
This method also hides the default button?

**Reply by Jacalz** [reply]
What default button? If you are talking about the submit button being disabled then no. You would have to use `form.SetOnValidationChanged()` for both forms and handle the disabled/enabled state of your own submit button.

**Reply by junmaqiang** [reply]
# this is my code
```
me := widget.NewEntry()
me.SetPlaceHolder("my name")

to := widget.NewEntry()
to.SetPlaceHolder("100")

formMe := &widget.Form{
		Items: []*widget.FormItem{
			{Text: "从", Widget: me, HintText: ""},
		},
		OnCancel: func() {
			fmt.Println("cong")
		},
		OnSubmit: func() {
			fmt.Println("cong Form submitted")
		},
}

formTo := &widget.Form{
  Items: []*widget.FormItem{
	  {Text: "到", Widget: to, HintText: ""},
  },
  OnCancel: func() {
	  fmt.Println("dao")
  },
  OnSubmit: func() {
	  fmt.Println("dao Form submitted")
  },
}

	con := container.NewHBox(formMe, formTo)

	fromLayout := container.NewBorder(nil, nil, nil, nil, container.NewStack(con))

	return fromLayout

```

# this is demo
<img width="389" alt="截屏2023-09-23 17 38 15" src="https://github.com/fyne-io/fyne/assets/1819457/968f9d82-792d-47a4-97aa-4584a1ac26e4">

## How to remove two default buttons？
Cancel  and Submit

**Reply by Jacalz** [reply]
Just don't set the OnCancel or OnSubmit fields in the form. Use your own buttons that you create and but to the right of the two forms, like I said above

**Reply by junmaqiang** [reply]
ok I see  ， It turns out that buttons are optional    thank you
